docs/fsharp/tutorials/asynchronous-and-concurrent-programming/async.md

```diff
@@ -1,213 +1,382 @@
+
+When you put these two terms together, you'll see that "asynchronous" means "not at the same time". That's it! There is no implication of concurrency or parallelism in this definition. This is also true in practice.
+
+In practical terms, asynchronous computations in F# are scheduled to execute independently of the main program flow. This doesn't imply concurrency or parallelism, nor does it imply that a computation always happens in the background. In fact, asynchronous computations could even execute synchronously, depending on the nature of the computation and the environment the computation is executing in.
+
+The main takeaway you should have is that asynchronous computations are independent of the main program flow. Although there are few guarantees about when or how an asynchronous computation executes, there are some approaches to orchestrating and scheduling them. The rest of this article explores core concepts for F# asynchrony and how to use the types, functions, and expressions built into F#.
+
+## Core concepts
+
+In F#, asynchronous programming is centered around three core concepts:
+
+- The `Async<'T>` type, which represents a composable asynchronous computation.
+- The `Async` module functions, which let you schedule asynchronous work, compose multiple asynchronous computations, and transform asynchronous results.
+- The `async { }` [computation expression](../../language-reference/computation-expressions.md), which provides a convenient syntax for describing and controlling asynchronous computations.
+
+You can see these three concepts in the following example:
 
 ```fsharp
 open System
-open System.Net
+open System.IO
 
-let fetchHtmlAsync url =
+let countFileBytes path =
     async {
-        let uri = Uri(url)
-        use webClient = new WebClient()
+        let! bytes = File.ReadAllBytesAsync(path) |> Async.AwaitTask
+        let fileName = Path.GetFileName(path)
+        printfn "File %s has %d bytes" fileName bytes.Length
+    }
+
+[<EntryPoint>]
+let main argv =
+    countFileBytes "path-to-file.txt"
+    |> Async.RunSynchronously
 
-        // Execution of fetchHtmlAsync won't continue until the result
-        // of AsyncDownloadString is bound.
-        let! html = webClient.AsyncDownloadString(uri)
-        return html
+    Console.Read() |> ignore
+    0 // return an integer exit code
+```
+
+In the example, the `countFileBytes` function is of type `string -> Async<unit>`. Calling the function does not actually execute the asynchronous computation. Instead, it returns an `Async<unit>` that acts as a *specification* of the work that is to execute asynchronously. It calls `Async.AwaitTask` in its body, which will convert the result of <xref:System.IO.File.WriteAllBytesAsync%2A> to an appropriate type when it is called.
+
+Another important line is the call to `Async.RunSynchronously`. This is one of the async starting functions that you'll need to call if you want to actually execute an F# asynchronous computation.
```

Comment:
>async starting 

Async starting - capital letter might mentally tie Async module to Async<T> in readers mind?

Reply:
Replaced with "Async module"